src/shared/types/models.ts

```diff
@@ -27,3 +27,71 @@ export interface JWTDataType {
   exp: string;
   type: string;
 }
+
+export interface ColumnFields {
+  title: string;
+  field: string;
+}
+
+export type DayIndexOptions = {
+  [key: string]: number;
+};
+
+export interface WeeklyAssignmentInterface {
+  time?: string;
+  duty: string;
+  [key: string]: string;
+}
+export interface MappedScheduleInterface {
+  day: string;
+  name: string;
+  columns: ColumnFields[];
+  data: WeeklyAssignmentInterface[];
+}
+
+export interface WeeklyAssignment {
+  [key: string]: string;
+}
+
+export interface TaskData {
+  date: string;
+  assignee: string;
+}
+export interface DutyData {
+  title: string;
+  tasks: TaskData[];
+  team?: string;
+}
+export interface EventData {
+  time: string;
+  duties: DutyData[];
+  tag?: string;
+}
+export interface WeeklyEventData {
+  day: string;
+  events: EventData[];
+  dividers: Divider[];
```

Comment:
for my understanding: what do we mean by a "divider"?

Reply:
so sometimes there are multiples services in a day: "Morning service, Afternoon service, evening service" etc. This divider helps us to differentiate those services.